### PR TITLE
ios: time based message grouping

### PR DESCRIPTION
**Pull request by levitatingpineapple**

### CODE REVIEW

File: apps/ios/Shared/Model/ChatModel.swift [hunk 1]
```diff
@@ -428,19 +436,17 @@ final class ChatModel: ObservableObject {
 
     private func _upsertChatItem(_ cInfo: ChatInfo, _ cItem: ChatItem) -> Bool {
         if let i = getChatItemIndex(cItem) {
-            withConditionalAnimation {
-                _updateChatItem(at: i, with: cItem)
-            }
```
Comment (levitatingpineapple) [inline apps/ios/Shared/Model/ChatModel.swift:433]: Two instances of `withConditionalAnimation ` removed.
- They haven't had effect since moving to reverse list
- They where interfering with chat item view update, when item below it causes the change

File: apps/ios/Shared/Views/Chat/ChatItemView.swift [hunk 2]
```diff
@@ -9,9 +9,21 @@
 import SwiftUI
 import SimpleXChat
 
+extension EnvironmentValues {
```
Comment (epoberezkin) [inline apps/ios/Shared/Views/Chat/ChatItemView.swift:12]: why is this needed

File: apps/ios/Shared/Views/Chat/ChatItemView.swift [hunk 3]
```diff
@@ -9,9 +9,21 @@
...
+    var showTimestamp: Bool {
+        get { self[ShowTimestamp.self] }
+        set { self[ShowTimestamp.self] = newValue }
+    }
+}
+
 struct ChatItemView: View {
     @ObservedObject var chat: Chat
     @EnvironmentObject var theme: AppTheme
+    @Environment(\.showTimestamp) var showTimestamp
```
Comment (epoberezkin) [inline apps/ios/Shared/Views/Chat/ChatItemView.swift:26]: why does it need to be Environment? Seems some fancy way to pass parameter.

File: apps/ios/Shared/Views/Chat/ChatView.swift [hunk 4]
```diff
@@ -716,31 +717,53 @@ struct ChatView: View {
 
         var revealed: Bool { chatItem == revealedChatItem }
 
+        struct TimeSeparation {
```
Comment (epoberezkin) [inline apps/ios/Shared/Views/Chat/ChatView.swift:720]: why is it a struct?

Comment (epoberezkin) [reply apps/ios/Shared/Views/Chat/ChatView.swift:720]: we are passing two parameters, any encapsulation beyond maybe ad hoc tuple, is not necessary here. Or just two parameters

File: apps/ios/Shared/Views/Chat/ChatView.swift [hunk 5]
```diff
@@ -716,31 +717,53 @@ struct ChatView: View {
...
+            let isTimestampShown: Bool
+            let isGapLarge: Bool
+
+            init(for chatItem: ChatItem, at index: Int?) {
```
Comment (epoberezkin) [inline apps/ios/Shared/Views/Chat/ChatView.swift:724]: inits are not supposed to do any work, they are supposed to be dummy constructor.

Making inits do some logic is OOP anti-pattern
